core/src/main/scala/chisel3/Bits.scala

```diff
@@ -778,11 +778,26 @@ sealed class UInt private[chisel3] (width: Width) extends Bits(width) with Num[U
   final def zext: SInt = macro SourceInfoTransform.noArg
 
   /** @group SourceInfoTransformMacro */
-  def do_zext(implicit sourceInfo: SourceInfo): SInt =
-    pushOp(DefPrim(sourceInfo, SInt(width + 1), ConvertOp, ref))
+  def do_zext(implicit sourceInfo: SourceInfo): SInt = this.litOption match {
+    case Some(value) => SInt.Lit(value, this.width + 1)
+    case None        => pushOp(DefPrim(sourceInfo, SInt(width + 1), ConvertOp, ref))
+  }
```

Comment:
It is somewhat surprising that zero extending a `UInt` returns an `SInt`. I would've thought that this took a width _to zero extend to and did not change the type_. This seems like it wants to be `asSIntPreserveValue` or something. This is what the API is, though.